### ISSUE TITLE
New version of bambulabs supports camera in lan mode

### ISSUE THREAD

**Issue by joeblack2k** [hisptoot/BambuSource2Raw#7]
is there a plan to update the code that would support the new mode?

https://github.com/bambulab/BambuStudio/releases/tag/v01.07.00.53

`Liveview in LAN only mode (X1) https://github.com/bambulab/BambuStudio/issues/1372
We have received multiple requests from our customers in regard to having liveview available in LAN only mode. The latest firmware for X1 series combined with the latest version of Bambu Studio introduces support for accessing the video from the printer locally, without requiring an internet connection.
Please note that when the printer is set in LAN only mode, you will not have access to the liveview stream from Bambu Handy.
Please update the printer firmware to V01.06 as well.`

**Comment by hisptoot** [comment]
Yes, I will try to add this mode when the new firmware come out.

**Comment by hisptoot** [comment]
The new firmware of x1 seems to use rtsp/rtsps to provide local video streaming. 
The url is rtsp/rtsps://bblp:[access_code]@[lan_ip]/streaming/live/1?device=[dev_id]&version=01.07.00.03

They seem to use Live555 in their libBambuSource module.
see https://github.com/cmguo/live555/

If my presumption is right, you can get rtsp stream directly from x1 and my program is not needed.

**Comment by joeblack2k** [comment]
oh wow! that is a big change and highly requested!

**Comment by hisptoot** [comment]
I just update the code for x1c local mode.

You can also use ffmpeg to get video stream directly from x1c:

ffmpeg -i rtsps://bblp:[access_code]@[ip_addr]/streaming/live/1 -c copy -y test.mkv

**Comment by su-sd** [comment]
Does the ffmpeg command also work for P1S?
On my machine I get
```
[ffmpeg Version and compile options]
[tcp @ 0x555a2d9de540] Connection to tcp://192.168.yyy.xxx:554?timeout=0 failed: Connection refused
[in#0 @ 0x555a2da29e00] Error opening input: Connection refused
Error opening input file rtsp://.../streaming/live/1.
Error opening input files: Connection refused
```
